GymDB/1 tables.sql

```diff
@@ -0,0 +1,24 @@
+use GymDB 
+go 
+
+drop table if exists Member 
+go
+
+create table dbo.Member(
+    MemberId int not null identity primary key,
+    FirstName varchar(50) not null constraint ck_Member_FirstName_cannot_be_blank check(FirstName <> ''),
+    LastName varchar(50) not null constraint ck_Member_LastName_cannot_be_blank check(LastName <> ''),
+    ContactNumber varchar(8) not null constraint ck_Member_ContactNumber_must_be_3_numbers_a_dash_then_4_numbers check(ContactNumber like '[0-9][0-9][0-9][-][0-9][0-9][0-9][0-9]'),
+    EmergencyContactNumber varchar(8) not null constraint ck_Member_EmergencyContactNumber_must_be_3_numbers_a_dash_then_4_numbers check(EmergencyContactNumber like '[0-9][0-9][0-9][-][0-9][0-9][0-9][0-9]'),
+    InitialWeight int not null constraint ck_Member_InitialWeight_must_be_greater_than_zero check(InitialWeight > 0),
+    GoalWeight int not null constraint ck_Member_GoalWeight_must_be_greater_than_zero check(GoalWeight > 0),
+    CurrentWeight int not null constraint ck_Member_CurrentWeight_must_be_greater_than_zero check(CurrentWeight > 0),
+    BMI decimal(4, 1) not null constraint ck_Member_BMI_must_be_greater_than_zero check(BMI > 0),
+    MembershipStartDate date not null constraint ck_Member_MembershipStartDate_must_be_after_december_1st_2022 check(MembershipStartDate >= '2022-12-01'),
+    MembershipEndDate date not null,
+    MembershipType varchar(20) not null constraint ck_Member_MembershipType_must_be_either_annual_semiannual_monthly_or_quarterly check(MembershipType in ('Annual', 'Semi-annual', 'Monthly', 'Quarterly')),
+    PaymentStatus bit not null,
+    SessionDateAndTime datetime not null,
+    constraint ck_Member_MembershipStartDate_is_before_MembershipEndDate check(MembershipStartDate < MembershipEndDate)
```

Comment:
I mentioned that I would like the **progress** of each member to be entered with every record and that I would like it to be auto generated.

Reply:
That is a bit too vague. Can you explain what you had in mind? I don't see a way of doing that.

Reply:
I would like a column that calculates the progress in weight loss with every session. I think you would have to subtract the current weight from the start weight as a constraint in a weight loss column.

Reply:
Ok, I added that in and fixed the corresponding queries to use this new column. I thought you meant you wanted the progress based off of previous records for that member, didn't realize you wanted for each record stand alone.